File: backend/spring/clean-architecture/minimal/application/src/main/java/apptemplate/application/mappers/UploadedFileMapper.java
package apptemplate.application.mappers;

import apptemplate.application.dto.file.UploadedFileDto;
import apptemplate.domain.entities.UploadedFile;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;
import org.mapstruct.factory.Mappers;

@Mapper(componentModel = "spring")
public interface UploadedFileMapper {

    UploadedFileMapper INSTANCE = Mappers.getMapper(UploadedFileMapper.class);

    @Mapping(target = "downloadUrl", expression = "java(\"/api/files/\" + file.getId() + \"/download\")")
    UploadedFileDto toDto(UploadedFile file);
}

File: backend/nestjs/clean-architecture/minimal/src/modules/file-management/application/mappers/uploaded-file.mapper.ts
import { UploadedFile } from '../../domain/entities/uploaded-file.entity';
import { UploadedFileDto } from '../dto/uploaded-file.dto';

export class UploadedFileMapper {
  static toDto(file: UploadedFile): UploadedFileDto {
    return {
      id: file.id,
      fileName: file.fileName,
      originalFileName: file.originalFileName,
      contentType: file.contentType,
      fileSize: file.fileSize,
      description: file.description,
      category: file.category,
      isPublic: file.isPublic,
      createdAt: file.createdAt,
      updatedAt: file.updatedAt,
      createdBy: file.createdBy,
      downloadUrl: `/api/files/${file.id}/download`,
    };
  }
}
